### DISCUSSION TITLE
Named snippets dynamically imported from external Git repo

### DISCUSSION THREAD

**Discussion by machty** [GitbookIO/community#231] [Feature Requests]
### What's your feature/idea?

A lot of Google's docs use code snippets that are extracted from an actual Git repository of code, e.g. [this page](https://developers.google.com/ml-kit/vision/barcode-scanning/android) in their MLKit docs uses snippets from [this file in a Github repo](https://github.com/googlesamples/mlkit/blob/e07a940d2a25d278c2a51842be159f7a7be747f1/android/android-snippets/app/src/main/java/com/google/example/mlkit/kotlin/BarcodeScanningActivity.kt#L28C1-L34).

They even allow multiple file names, e.g. Kotlin and Java, to declare the same snippet name, and the code snippets are grouped together in a language selector tab:

<img width="893" alt="image" src="https://github.com/GitbookIO/community/assets/81818/ccbd8a1e-9433-4bf2-aea4-52660a32a1fe">

In order for this to work, you have to put START/END comments in your codebase with the name of the snippet. There are many benefits to this:

1. It's much more different for documentation on Gitbook to become stale; code snippet blocks, so long as they're not deleted, prove robust against line number changes, and even file number changes
1. As a corollary you don't even need to specify the file name where a code snippet lives, just the name of the code snippet
1. Anyone reading the code will see the snippet and know that there is some documentation on Gitbook (or wherever) that references it, which improves documentation discoverability

### What's your desired outcome?

Gitbook provides some feature by which an external database that serves as a snippet source can be linked to the account, and adds UI for adding snippets to gitbook docs via snippet name.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by jayonsoftware** [comment]
What is the best way to insert code snippets into Gitbook while maintaining an ever-changing code base? Currently, I copy and paste and make mistakes. Can something like this be done?

```
// [START name, lang]
.... Your Code
// [END name, lang]
```

Where the name is a unique name for the block, and lang is the programming lang the code is in

And on Gitbook, I should be able to paste something like 

`https://github.com/jayonsoftware/GpuTest/blob/main/Program.cs?codeblock=name`

And this will provide the same function as <CodeBlock>, but the code will be synced to git. 

Thoughts ?

Jay
